src/Entity/Donation.php

```diff
@@ -636,4 +636,18 @@ public function isSuccess(): bool
     {
         return \in_array($this->status, [self::STATUS_FINISHED, self::STATUS_SUBSCRIPTION_IN_PROGRESS], true);
     }
+
+    public function isFirstSuccessfulTransaction(Transaction $transaction): bool
+    {
+        if (!$transaction->isSuccessful()) {
+            return false;
+        }
+
+        $successfulTransactions = array_filter(
+            $this->transactions->toArray(),
+            fn (Transaction $t) => $t->isSuccessful()
+        );
+
+        return 1 === \count($successfulTransactions) && $this->transactions->first() === $transaction;
```

Comment:
we should compare with the first transaction of `$successfulTransactions`, no ?

```suggestion
        return 1 === \count($successfulTransactions) && current($successfulTransactions) === $transaction;
```